### PR TITLE
Update E0424 to the new error format

### LINKED ISSUE #35797: E0424 needs to be updated to new format
From: src/test/compile-fail/E0424.rs

E0424 needs a span_label, a note, and an updated title, changing it from:

```
error[E0424]: `self` is not available in a static method. Maybe a `self` argument is missing?
  --> src/test/compile-fail/E0424.rs:17:9
   |
17 |         self.bar(); //~ ERROR E0424
   |         ^^^^
```

To:

```
error[E0424]: `self` is not available in a static method
  --> src/test/compile-fail/E0424.rs:17:9
   |
17 |         self.bar(); //~ ERROR E0424
   |         ^^^^ not available in static method
   |
   = note: maybe a `self` argument is missing?
```

### PR DESCRIPTION
**Pull request by kyrias**
Fixes #35797.
Part of #35233.

r? @GuillaumeGomez

### CODE REVIEW

File: src/test/compile-fail/issue-2356.rs [hunk 1]
```diff
@@ -59,7 +59,7 @@ impl cat {
 impl cat {
   fn meow() {
     if self.whiskers > 3 {
-        //~^ ERROR: `self` is not available in a static method. Maybe a `self` argument is missing?
+        //~^ ERROR: `self` is not available in a static method
```
Comment (GuillaumeGomez) [inline src/test/compile-fail/issue-2356.rs]: Missing the note you added.

**Comment by kyrias** [comment]
Ah, added that afterwards and forgot to check the diff of the first file.

**Comment by GuillaumeGomez** [comment]
No problem. Please squash your commits and then I r+.

**Comment by kyrias** [comment]
Squashed!

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**